mlbench_core/optim/pytorch/optim.py

```diff
@@ -499,12 +499,302 @@ def step(self, closure=None):
         return loss
 
 
+class PowerSGD(Optimizer):
+    def __init__(
+        self,
+        model=None,
+        lr=0.1,
+        weight_decay=0,
+        momentum=0,
+        dampening=0,
+        nesterov=False,
+        reuse_query=False,
+        rank=1,
+    ):
+        if not model:
+            raise ValueError('"model" not set for optimizer')
+        if lr is not required and lr < 0.0:
```

Comment:
I think it would be nicer to separate the optimizer and the aggregator (PowerSGD w/ EF), so you can plug it in instead of all-reduce into any optimizer (Adam, Adagrad, ...)